Added hourly interest for Aave on Base

mikeghen opened the pull request:
## Thank you for contributing to Spellbook 🪄
Please open the PR in **draft** and mark as ready when you want to request a review. 

### Description:
* Added hourly interest for Aave on Base. 
* Follows the existing implementation implementation for Aave interest rates on Optimism. 


---
quick links for more information:
- [README.md](https://github.com/duneanalytics/spellbook/blob/main/README.md)
- [spellbook docs](https://github.com/duneanalytics/spellbook/tree/main/docs)
- [CONTRIBUTING.md](https://github.com/duneanalytics/spellbook/blob/main/CONTRIBUTING.md)

mikeghen commented:
I have read the CLA Document and I hereby sign the CLA.

Hosuke commented on dbt_subprojects/daily_spellbook/models/aave/base/aave_base_schema.yml:
```diff
@@ -54,3 +54,53 @@ models:
+      contributors: mikeghen1, batwayne, chuxin
+    config:
+      tags: ['base', 'aave_v3', 'interest', 'aave', 'lending']
+    description: "Hourly aave interest rates on base"
+    columns:
```
```suggestion
    description: "Hourly aave interest rates on base"
    data_tests:
      - dbt_utils.unique_combination_of_columns:
          combination_of_columns:
            - reserve
            - symbol
            - hour
    columns:
```

Hosuke commented on dbt_subprojects/daily_spellbook/models/aave/base/aave_v3_base_interest_rates.sql:
```diff
@@ -0,0 +1,28 @@
+{{ config(
+   schema = 'aave_v3_base'
+  , materialized = 'incremental'
+  , file_format = 'delta'
+  , incremental_strategy = 'merge'
+  , unique_key = ['reserve', 'symbol', 'hour']
+  , alias = 'interest'
+  , post_hook='{{ expose_spells(blockchains = \'["base"]\',
```
```suggestion
   schema = 'aave_v3_base'
  , alias = 'interest_rates'
  , materialized = 'incremental'
  , file_format = 'delta'
  , incremental_strategy = 'merge'
  , unique_key = ['reserve', 'symbol', 'hour']
  , post_hook='{{ expose_spells(blockchains = \'["base"]\',
```

Hosuke commented on dbt_subprojects/daily_spellbook/models/aave/base/aave_v3_base_interest_rates.sql:
```diff
@@ -0,0 +1,28 @@
+  avg(CAST(a.variableBorrowRate AS DOUBLE)) / 1e27 as variable_borrow_apy
+from {{ source('aave_v3_base', 'Pool_evt_ReserveDataUpdated') }} a
+left join {{ source('tokens_base', 'erc20') }} t
+on a.reserve = t.contract_address
```
`symbol` is currently null.

mikeghen replied:
I changed to use `tokens.erc20`: https://github.com/duneanalytics/spellbook/pull/7404/commits/ceb747b6922de227703938a8fe5a0c9aca2f9dee

Hosuke commented on dbt_subprojects/daily_spellbook/tests/aave/base/aave_base_interests_test.sql:
```diff
@@ -0,0 +1,25 @@
+                 else false
+               end as test
+        from   {{ ref('aave_v3_base_interest_rates' )}}
+        where  symbol = 'USDC'
```
This test may not work properly.

mikeghen replied:
I updated this test to use correct values and a valid token symbol: https://github.com/duneanalytics/spellbook/pull/7404/commits/2f80416f3c5d351218f2d34fa4c048f1e9110fc3

mikeghen replied:
I updated these values to use real ones and the token symbol is now WETH instead: https://github.com/duneanalytics/spellbook/pull/7404/commits/2f80416f3c5d351218f2d34fa4c048f1e9110fc3

Hosuke commented:
Current [aave_v3_base_interest_rates](https://dune.com/queries/4510183)

mikeghen commented:
@Hosuke Thank you for the review, and for sharing a link to the table, I was having trouble finding the table in the test_schema. I'll work on fixing the issues and let you know when it's ready for another review.

mikeghen commented:
@Hosuke, I addressed the comments you left above if you would check they are resolved. 

Also, I was wondering if you might know why the table created has it's most recent data as "2023-11-11 08:00," over a year ago. Is this because of the way `test_schema` works? I would have expected dates as recent as today 2025-01-02.

Hosuke commented:
> @Hosuke, I addressed the comments you left above if you would check they are resolved.
> 
> Also, I was wondering if you might know why the table created has it's most recent data as "2023-11-11 08:00," over a year ago. Is this because of the way `test_schema` works? I would have expected dates as recent as today 2025-01-02.

The reason is the source you used `aave_v3_base.Pool_evt_ReserveDataUpdated` only has the latest data at `2023-11-11 08:54`:
https://dune.com/queries/4512697

We may need to verify this source, or use/submit another better contract's evt.

mikeghen commented:
@Hosuke, I took a look and I think I found the right table to use: `aave_v3_base.L2Pool_evt_ReserveDataUpdated` which looks like the current deployment of the Aave Pool contract, the other table is the an old pool contract I think.

Hosuke reviewed (approved):
LGTM.✅
Thank you @mikeghen 

[Sample aave_v3_base_interest_rates](https://dune.com/queries/4514285)

mikeghen commented:
Great, thanks for your help @Hosuke 🙏

jeff-dude reviewed (approved):
thanks for joining us and building in spellbook!
